runtime/src/bank.rs

```diff
@@ -659,16 +668,17 @@ impl Bank {
             .iter()
             .zip(executed.iter())
             .map(|(tx, res)| {
+                let fee = self.fee_calculator.calculate_fee(tx.message());
                 let message = tx.message();
                 match *res {
                     Err(TransactionError::InstructionError(_, _)) => {
                         // Charge the transaction fee even in case of InstructionError
-                        self.withdraw(&message.account_keys[0], message.fee)?;
-                        fees += message.fee;
+                        self.withdraw(&message.account_keys[0], fee)?;
```

Comment:
we only have to do this directly on a program error?  withdrawing the fee from the sender is otherwise done by the runtime, the program, or ... ?

Reply:
@pgarg66, implemented this. I'm fairly sure it needs a closer look. I'd think that we should be collecting a fee for all except some of the InsufficientFunds errors.  Also, I don't know if we should be returning `Ok(())` in this situation. Seems weird to me and I've tripped up on it before in unit-tests.

Reply:
I think the alternative to making it Ok(()) here is to propagate the InstructionError logic around, so it's a toss-up